internal/server/place/places.go

```diff
@@ -207,53 +207,66 @@ func GetPlaceMetadataHelper(
 		return nil, err
 	}
 	result := map[string]*pb.PlaceMetadata{}
+	metaMap := map[string]*pb.PlaceMetadataCache_PlaceInfo{}
 	for _, btData := range btDataList {
 		for _, row := range btData {
-			entity := row.Parts[0]
-			if _, ok := result[entity]; ok {
-				continue
-			}
 			raw, ok := row.Data.(*pb.PlaceMetadataCache)
 			if !ok {
 				continue
 			}
-			processed := pb.PlaceMetadata{}
-			metaMap := map[string]*pb.PlaceMetadataCache_PlaceInfo{}
 			for _, info := range raw.Places {
-				metaMap[info.Dcid] = info
+				if currInfo, ok := metaMap[info.Dcid]; !ok {
+					metaMap[info.Dcid] = info
+				} else {
+					// Merge place info from new import group.
+					if currInfo.Name == "" {
+						currInfo.Name = info.Name
+					}
+					if currInfo.Type == "" {
+						currInfo.Type = info.Type
+					}
```

Comment:
Something to think about...:
Wonder if it's possible to have   currInfo.Type != info.Type? In that case, shall we raise an error? Or prefer the main DC one?
Similar comment for curInfo.Name above.

Reply:
For all the merging, we rank custom DC higher, and the loop here is from most ranked import group. So now if higher ranked import has empty value, then use the lower ranked one. If the values conflict, then still prefer the higher ranked import group..